src/hooks/use-prompt-usage.tsx

```diff
@@ -0,0 +1,90 @@
+import * as Sentry from '@sentry/electron/renderer';
+import { useState, useEffect, useCallback, useMemo, createContext, useContext } from 'react';
+import { LIMIT_OF_PROMPTS_PER_USER } from '../constants';
+import { useAuth } from './use-auth';
+import { getAppGlobals } from '../lib/app-globals';
+
+type PromptUsage = {
+	promptLimit: number;
+	promptCount: number;
+	fetchPromptUsage: () => Promise< void >;
+};
+
+const initState = {
+	promptLimit: LIMIT_OF_PROMPTS_PER_USER,
+	promptCount: 0,
+	fetchPromptUsage: async () => undefined,
+};
+const promptUsageContext = createContext< PromptUsage >( initState );
+
+interface PromptUsageProps {
+	children?: React.ReactNode;
+}
+
+export function usePromptUsage() {
+	return useContext( promptUsageContext );
+}
+
+export function PromptUsageProvider( { children }: PromptUsageProps ) {
+	const { Provider } = promptUsageContext;
+	const assistantEnabled = getAppGlobals().assistantEnabled;
+
+	const [ promptLimit, setPromptLimit ] = useState( LIMIT_OF_PROMPTS_PER_USER );
+	const [ promptCount, setPromptCount ] = useState( 0 );
+	const { client } = useAuth();
+
+	const fetchPromptUsage = useCallback( async () => {
+		if ( ! client?.req || ! assistantEnabled ) {
+			return;
+		}
+		try {
+			await new Promise( ( resolve, reject ) => {
+				client.req.get(
+					{
+						method: 'HEAD',
+						path: '/studio-app/ai-assistant/chat',
+						apiNamespace: 'wpcom/v2',
+					},
+					( error: Error, _data: unknown, headers: Record< string, string > ) => {
+						if ( error ) {
+							return reject( error );
+						}
+						if ( ! headers ) {
+							reject( new Error( 'No headers in response' ) );
+							return;
+						}
+						const limit = parseInt( headers[ 'x-ratelimit-limit' ] );
+						const remaining = parseInt( headers[ 'x-ratelimit-remaining' ] );
+						if ( isNaN( limit ) || isNaN( remaining ) ) {
+							reject( new Error( 'Error fetching limit response' ) );
+							return;
+						}
+						setPromptLimit( limit );
+						setPromptCount( limit - remaining );
+						resolve( headers );
+					}
+				);
+			} );
+		} catch ( error ) {
+			Sentry.captureException( error );
+			console.error( error );
+		}
+	}, [ client ] );
```

Comment:
Should we add `assistantEnabled` to the dependency array?

I'm getting this lint warning:


<img width="1241" alt="Screenshot 2024-06-04 at 11 19 57" src="https://github.com/Automattic/studio/assets/779993/8d7c33e4-46c6-4152-9919-cea7c8d5c346">



```suggestion
	}, [ assistantEnabled, client ] );
```